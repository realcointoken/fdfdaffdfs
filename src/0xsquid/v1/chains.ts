import { removeEmpty } from "./util";
import {
  BaseChain,
  ChainData,
  ChainsResponse,
  ChainType,
  CosmosChain,
  EvmChain
} from "../../types";

export const parseChainNativeContracts = (
  data: any
): {
  wrappedNativeToken: string;
  ensRegistry: string;
  multicall: string;
  usdcToken: string;
} => {
  const { wrappedNativeToken, ensRegistry, multicall, usdcToken } = data;
  return { wrappedNativeToken, ensRegistry, multicall, usdcToken };
};

export const parseAxelarContracts = (
  data: any
): {
  gateway: string;
  forecallable?: string;
} => {
  const { gateway, forecallable } = data;
  return removeEmpty({ gateway, forecallable });
};

export const parseSquidContracts = (
  data: any
): {
  squidRouter: string;
  defaultCrosschainToken?: string;
  squidMulticall?: string;
} => {
  const { squidRouter, defaultCrosschainToken, squidMulticall } = data;
  return removeEmpty({ squidRouter, defaultCrosschainToken, squidMulticall });
};

export const parseBaseChain = (data: any): BaseChain => {
  const {
    chainName,
    chainType,
    rpc,
    internalRpc,
    networkName,
    chainId,
    nativeCurrency,
    chainIconURI,
    blockExplorerUrls,
    axelarContracts,
    squidContracts,
    estimatedRouteDuration,
    estimatedExpressRouteDuration
  } = data;
  return {
    chainName,
    chainType,
    rpc,
    internalRpc,
    networkName,
    chainId,
    nativeCurrency,
    chainIconURI,
    blockExplorerUrls,
    axelarContracts: parseAxelarContracts(axelarContracts),
    squidContracts: parseSquidContracts(squidContracts),
    estimatedRouteDuration,
    estimatedExpressRouteDuration
  } as BaseChain;
};

export const parseEvmChain = (data: any): EvmChain => {
  const baseChain = parseBaseChain(data);
  const { chainNativeContracts } = data;
  return {
    ...baseChain,
    chainNativeContracts: parseChainNativeContracts(chainNativeContracts)
  } as EvmChain;
};

export const parseCosmosChain = (data: any): CosmosChain => {
  const baseProperties = parseBaseChain(data);
  const {
    rest,
    stakeCurrency,
    walletUrl,
    walletUrlForStaking,
    bip44,
    alternativeBIP44s,
    bech32Config,
    currencies,
    feeCurrencies,
    coinType,
    features,
    gasPriceStep,
    chainToAxelarChannelId
  } = data;

  return removeEmpty({
    ...baseProperties,
    rest,
    stakeCurrency,
    walletUrl,
    walletUrlForStaking,
    bip44,
    alternativeBIP44s,
    bech32Config,
    currencies,
    feeCurrencies,
    coinType,
    features,
    gasPriceStep,
    chainToAxelarChannelId
  }) as CosmosChain;
};

export const parseChainData = (data: any[]): ChainData[] => {
  const chains = data
    .filter((chain: BaseChain) =>
      [ChainType.EVM, ChainType.Cosmos].includes(chain.chainType)
    )
    .map((chain: any) => {
      switch (chain.chainType as ChainType) {
        case ChainType.EVM:
          return parseEvmChain(chain);
        case ChainType.Cosmos:
          return parseCosmosChain(chain);
      }
    });
  return chains;
};

export const parseChainsResponse = (response: any): ChainsResponse => {
  const chainsResponse = {
    chains: parseChainData(response.chains)
  };
  return chainsResponse;
};
