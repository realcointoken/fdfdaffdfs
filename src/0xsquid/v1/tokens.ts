import { TokenData, TokensResponse } from "../../types";

export const parseTokenData = (data: any): TokenData => {
  const { chainId, address, name, symbol, decimals, logoURI, coingeckoId } =
    data;
  return {
    chainId,
    address,
    name,
    symbol,
    decimals,
    logoURI,
    coingeckoId
  };
};

export const parseTokenDataList = (data: any[]): TokenData[] => {
  const tokenDataList = data.map((token: any) => {
    return parseTokenData(token);
  });
  return tokenDataList;
};

export const parseTokensResponse = (response: any): TokensResponse => {
  const tokensResponse = {
    tokens: parseTokenDataList(response.tokens)
  };
  return tokensResponse;
};
