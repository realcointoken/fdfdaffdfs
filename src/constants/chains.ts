import { ChainName, MapChainIdName } from "../types";

export const mapChainIdName: MapChainIdName = {
  1: ChainName.ETHEREUM,
  3: ChainName.ETHEREUM,
  56: ChainName.BINANCE,
  97: ChainName.BINANCE,
  250: ChainName.FANTOM,
  4002: ChainName.FANTOM,
  1313161554: ChainName.AURORA,
  1313161555: ChainName.AURORA,
  43113: ChainName.AVALANCHE,
  43114: ChainName.AVALANCHE,
  1287: ChainName.MOONBEAM,
  1284: ChainName.MOONBEAM,
  137: ChainName.POLYGON,
  80001: ChainName.POLYGON,
  cosmoshub: ChainName.COSMOS,
  "osmo-test-4": ChainName.OSMOSIS4,
  "osmosis-4": ChainName.OSMOSIS,
  "sei-testnet-1": ChainName.SEI,
  "fetchhub-3": ChainName.FETCH,
  "crescent-1": ChainName.CRESCENT,
  "emoney-3": ChainName.EMONEY,
  "injective-1": ChainName.INJECTIVE,
  "juno-1": ChainName.JUNO,
  "kaiyo-1": ChainName.KUJIRA,
  "harpoon-4": ChainName.KUJIRA,
  "secret-4": ChainName.SECRET,
  "terra-2": ChainName.TERRA2,
  "phoenix-1": ChainName.TERRA2,
  "axelar-testnet-lisbon-3": ChainName.AXELARNET,
  "cosmoshub-4": ChainName.COSMOS
};
