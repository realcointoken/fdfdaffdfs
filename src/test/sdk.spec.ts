import axios from "axios";

import { Squid } from "../index";
import { chainsData } from "./constants/chains";
import { supportedTokens } from "./constants/tokens";

jest.mock("axios");

describe("Squid", () => {
  it("should instance Squid as expected", async () => {
    const getMocked = jest.fn().mockResolvedValue({
      data: {
        chains: chainsData,
        tokens: supportedTokens
      },
      status: 200
    });
    const mockedAxios = (axios.create as jest.Mock).mockReturnValue({
      get: getMocked,
      interceptors: {
        response: {
          use: jest.fn()
        }
      }
    });
    const squidSdk = new Squid();

    expect(squidSdk).toMatchSnapshot();
    expect(squidSdk.initialized).toBe(false);
    expect(squidSdk.tokens).toEqual([]);
    expect(squidSdk.chains).toEqual([]);
  });

  describe("setConfig", () => {
    it("should set config as expected", () => {
      const squidSdk = new Squid();

      squidSdk.setConfig({
        executionSettings: {
          infiniteApproval: false
        }
      });

      expect(squidSdk.config).toStrictEqual({
        executionSettings: {
          infiniteApproval: false
        }
      });
    });
  });

  describe("axios module", () => {
    it("get", async () => {
      const getMocked = jest.fn().mockResolvedValue({
        data: {
          chains: chainsData,
          tokens: supportedTokens
        },
        status: 200
      });
      const mockedAxios = (axios.create as jest.Mock).mockReturnValue({
        get: getMocked,
        interceptors: {
          response: {
            use: jest.fn()
          }
        }
      });

      const squidSdk = new Squid();

      await squidSdk.init();
      expect(squidSdk.initialized).toEqual(true);
    });
  });
});
