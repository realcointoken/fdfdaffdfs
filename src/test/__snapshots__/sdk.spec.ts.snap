// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Squid should instance Squid as expected 1`] = `
Squid {
  "axiosInstance": Object {
    "get": [MockFunction],
    "interceptors": Object {
      "response": Object {
        "use": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
    },
  },
  "chains": Array [],
  "config": Object {
    "baseUrl": "https://testnet.api.0xsquid.com/",
  },
  "initialized": false,
  "isInMaintenanceMode": false,
  "tokens": Array [],
}
`;
